projects/plugins/jetpack/extensions/blocks/tiled-gallery/gallery-image/withProxy.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * WordPress dependencies
+ */
+import { createHigherOrderComponent } from '@wordpress/compose';
+import { useState, useEffect } from '@wordpress/element';
+import apiFetch from '@wordpress/api-fetch';
+import { addQueryArgs } from '@wordpress/url';
+
+const cache = {};
+const cacheResponse = ( requestId, blob, freshness ) => {
+	// Cache at most 100 items
+	const cacheKeys = Object.keys( cache );
+	if ( cacheKeys.length > 100 ) {
+		delete cache[ cacheKeys[ 0 ] ];
+	}
+
+	cache[ requestId ] = blob;
+
+	// Self-remove this entry after `freshness` ms
+	setTimeout( () => {
+		delete cache[ requestId ];
+	}, freshness );
+};
+
+const withProxy = WrappedComponent => props => {
```

Comment:
Maybe we don't even need the hooks here – see the excellent suggestions @dmsnell left here: https://github.com/Automattic/wp-calypso/commit/b854bfd3a8c0c1f42551cc9fc5aaa9350ff46bee#commitcomment-38349034